src/sentry/api/endpoints/project_details.py

```diff
@@ -466,7 +466,8 @@ def put(self, request: Request, project) -> Response:
         old_data = serialize(project, request.user, DetailedProjectSerializer())
         has_project_write = request.access and (
             request.access.has_scope("project:write")
-            or request.access.has_project_scope(project, "project:write")
+            or request.access.has_scope("project:admin")
+            or request.access.has_any_project_scope(project, ["project:write", "project:admin"])
```

Comment:
`has_scope` is org level `project:admin`. so for instance an org admin has `project:admin` at the org level which means they can make modifications to projects even if they aren't on the same team as a project.

`has_any_project_scope` is looking for if the member has any of the scopes in the list passed into it at the project level. this is for team admins because they get scopes on the project level depending on if they are on the same team as a project. e.g. if someone is a team admin on some team, they will have `project:write` for all projects associated with that team

Reply:
the version of `has_any_project_scope` used depends on the type of auth token was used. I believe user and org tokens will use the DbAccess class `has_any_project_scope` tho so that's the only one that matters